public/flightControl/src/components/pre-flight-panel/FileSelector.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+interface FileSelectProps {
+	filenames: Array<string>,
+	handleChange(newValue: string): void
+}
+
+interface FileSelectState {
+	value: string
+}
+
+export class FileSelector extends React.Component<FileSelectProps, FileSelectState> {
```

Comment:
IMO if the react component doesn't really do anything but wrap an input, you probably don't need an extra component. But, either way, this component does not need to have it's own state. Use getDefaultProps instead. Having the state also prevents validation by the parent.